### ISSUE TITLE
Hostnames are fiddly in docker-compose

### ISSUE THREAD

**Issue by Dustpancake** [vernemq/docker-vernemq#232]
**TL;DR** `DOCKER_VERNEMQ_DISCOVERY_NODE` cannot resolve with docker hostnames.

### Issue

I've been trying to create a scalable VerneMQ cluster using docker-compose for an IoT project, but have been struggling to get VerneMQ clustering to work successfully with the docker assigned hostnames.

Consider my example:
```yml
# docker-compose.yml
version: "3.7"
services:
  vmq0:
    image: vernemq/vernemq
    environment:
      DOCKER_VERNEMQ_ACCEPT_EULA: "yes"
      
  vmq:
    image: vernemq/vernemq
    depends_on:
      - vmq0
    environment:
      DOCKER_VERNEMQ_DISCOVERY_NODE: "vmq0"
      DOCKER_VERNEMQ_ACCEPT_EULA: "yes"
```
This fails since I believe Erlang host resolution requires full hostnames, i.e. some `xxx.yyy`. I infer this from the logs:
```
...
vmq_1   | 14:37:15.987 [info] cluster event handler 'vmq_cluster' registered
vmq_1   | 14:37:16.935 [info] Sent join request to: 'VerneMQ@vmq0'
vmq_1   | 14:37:16.939 [error] ** System running to use fully qualified hostnames **
vmq_1   | ** Hostname vmq0 is illegal **
vmq_1   | 14:37:16.944 [info] Unable to connect to 'VerneMQ@vmq0'
```

I found online a SO answer (which of course I have subsequently lost) for a docker-compose RabbitMQ solution, where they used a network alias for the root node adding the extension `.mqtt`; translating this for the VerneMQ cluster, I add the alias to the `vmq0` service, and then set the env variable `DOCKER_VERNEMQ_DISCOVERY_NODE="vmq0.mqtt"`. The logs look a little more promising:
```
...
vmq_1   | 14:50:27.316 [info] cluster event handler 'vmq_cluster' registered
vmq_1   | 14:50:28.278 [info] Sent join request to: 'VerneMQ@vmq0.mqtt'
vmq_1   | 14:50:28.291 [info] Unable to connect to 'VerneMQ@vmq0.mqtt'
```
I also found [this RabbitMQ](https://stackoverflow.com/questions/25500629/rabbitmq-cluster-does-not-work-on-ec2) SO answer, where the accepted answer states for hostname resolution

> Windows/Erlang/WhoKnows wants caps (probably because of the NetBIOS of olden days). 

I tried that. It also didn't work.

### Use-case solution

My workaround solution for now is just to use static IPs
```
# docker-compose.yml
version: "3.7"
services:
  vmq0:
    image: vernemq/vernemq
    environment:
      DOCKER_VERNEMQ_ACCEPT_EULA: "yes"
    networks:
      backend:
        ipv4_address: 172.18.5.2

  vmq:
    image: vernemq/vernemq
    depends_on:
      - vmq0
    environment:
      DOCKER_VERNEMQ_DISCOVERY_NODE: "172.18.5.2"
      DOCKER_VERNEMQ_ACCEPT_EULA: "yes"
    networks:
      - backend

networks:
  backend:
    ipam:
      config:
        - subnet: 172.18.5.0/24
```
Which produces logs
```
...
vmq_1   | 14:57:42.039 [info] cluster event handler 'vmq_cluster' registered
vmq_1   | 14:57:42.979 [info] Sent join request to: 'VerneMQ@172.18.5.2'
vmq0_1  | 14:57:42.990 [info] cluster node 'VerneMQ@172.18.5.3' UP
vmq_1   | 14:57:42.990 [info] cluster node 'VerneMQ@172.18.5.2' UP
...
vmq0_1  | 14:57:44.001 [info] successfully connected to cluster node 'VerneMQ@172.18.5.3'
vmq_1   | 14:57:44.003 [info] successfully connected to cluster node 'VerneMQ@172.18.5.2'
```
This is okay, but doesn't feel particularly elegant -- additionally, when scaling the cluster, if you have other services in the compose file, the DHCP assignment means you won't know their IP addresses until the cluster is started. A mitigation is either assigning static IP addresses for all services (which sort of undermines the docker hostname allocation), or to have a very rigid `depends_on:` chain, which makes tampering a little unreliable.

Is there something obvious I am missing? Has anyone found a solution to this issue? I couldn't find any discussion in previous issues, so decided to open this one.

Thanks in advance for taking a look!

**Comment by stefanorumi7** [comment]
@fjebaker  I am having the same issue, thanks for posting your workaround!

**Comment by Shah-Panam** [comment]
Add ".local" at the end of hostname an write hotname in block letters.
So your node name will be VerneMQ@VMQ0.local (sort of how FQDN is shown in Windows Server).

This worked for me.
![image](https://user-images.githubusercontent.com/67643848/172064890-989117ce-3b3c-4b21-acec-6af9dabe7256.png)
